csharp-basic-topics/ContainsMethodInCsharp/ContainsMethodTests/Tests.cs

```diff
@@ -0,0 +1,80 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Linq;
+
+namespace ContainsMethodTests
+{
+    [TestClass]
+    public class Tests
+    {
+        [TestMethod]
+        public void WhenStringValueExistInStringArray_ThenReturnBooleanValue()
+        {
+            var isExist = false;
+            var countries = new string[] { "USA", "England", "Germany", "Sweeden", "Poland" };
+
+            if (countries.Contains("Sweeden"))
+                isExist = true;
+            else
+                isExist = false;
+
+            Assert.IsTrue(isExist);
+        }
+
+        [TestMethod]
+        public void WhenStringValueExistInStringArray_ThenReturnStringValueOfExistedElement()
+        {
+            var articles = new string[] { "article-1", "article-2", "article-3", "article-4", "article-5" };
+
+            var isFounded = from article in articles
+                            where article.Contains("article-1")
+                            select article;
+
+            foreach (var found in isFounded)
+            {
+                Assert.AreEqual("article-1", found);
+            }
+        }
+
+        [TestMethod]
+        public void WhenGivenStringValueExistInText_ThenReturnTupleAsStarterPositionOfValueAndBooleanValue()
+        {
+            var result = find("Code-Maze", "If you want to read great articles, then let's check the Code-Maze.");
+            Assert.IsTrue(result.Item2);
+        }
+
+        static (int, bool) find(string givenString, string templateText)
+        {
```

Comment:
The find() method shouldn't be in the test class itself. It should be in a separate class and called from the test class when needed.

Reply:
This way you've duplicated the code, and haven't tested the method you're actually calling when running the application,